core/src/main/scala/org/dbpedia/extraction/mappings/TemplateMapping.scala

```diff
@@ -6,180 +6,231 @@ import org.dbpedia.extraction.transform.Quad
-          graph += new Quad(context.language, typeDataset, uri, propertyRdfType, cls.uri, node.sourceIri+"&mappedTemplate="+node.title.encoded)
+        // Depending on the following conditions we create a new "blank node" or append the data to the main resource.
+        // Example case for creating new resources are the pages: enwiki:Volkswagen_Golf , enwiki:List_of_Playboy_Playmates_of_2012
+        // Example case we could append to existing class are where we have to different mapped templates that one is a subclass of the other
+
+        // Condition #1
+        //  Check if the root template has been mapped to the corresponding Class of this template
+        //  If the mapping already defines a corresponding class & propery then we should create a new resource
+        var condition1_createCorrespondingProperty = correspondingClass != null &&
+          correspondingProperty != null && pageClass.relatedClasses.contains(correspondingClass)
+
+        // Condition #2
+        // If we have more than one of the same template it means that we want to create multiple resources. See for example
+        // the pages: enwiki:Volkswagen_Golf , enwiki:List_of_Playboy_Playmates_of_2012
+        val pageTemplateSet = pageNode.getAnnotation(TemplateMapping.TEMPLATELIST_ANNOTATION).getOrElse(Seq.empty)
+        val condition2_template_exists = pageTemplateSet.contains(node.title.decoded)
+        if (!condition2_template_exists) {
+          node.setAnnotation(TemplateMapping.TEMPLATELIST_ANNOTATION, pageTemplateSet ++ Seq(node.title.decoded))
         }
-    }
 
-    /**
-     * Generates a new URI from a template node
-     *
-     * @param subjectUri The base string of the generated URI
-     * @param templateNode The template for which the URI is to be generated
-     * @return The generated URI
-     */
-    private def generateUri(subjectUri : String, templateNode : TemplateNode) : String =
-    {
-        val properties = templateNode.children
+        // Condition #3
+        // The current mapping is a subclass or a superclass of previous class or owl:Thing
+        val condition3_subclass = mapToClass.relatedClasses.contains(pageClass) || pageClass.relatedClasses.contains(mapToClass) || mapToClass.equals(classOwlThing) || pageClass.equals(classOwlThing)
+
+
+        // Condition #4
+        //if we have more than one info boxes and name property of infobox is different than title of page than infobox belong to different
+        //entity and need to be saved as new resource.
+        //This is to avoid assigning of one entity properties to another entity because a single page can have multiple info boxes and each
+        //about different entity, see dbr:Helene_Demuth
 
-        //Cannot generate URIs for empty templates
-        if(properties.isEmpty)
+        //checking if node is an infobox
+        var isInfobox = false
```

Comment:
You can also avoid making `isInfobox` mutable by rewriting in this way:
```
        val isInfobox = if (node.title.decoded.contains("Infobox")) {
          true
        } else {
          false
        }
```

Reply:
resolved